enable_disable_resources.md

```diff
@@ -0,0 +1,366 @@
+KubernetesApply controller consumes that status update and toggles the
+enabled/disabled state.
+
+The UIResource controller aggregates all the DisableOnStatus fields for a resource,
+and presents that state to the UI.
+
+```go
+type UIResourceStatus struct {
+  Enabled UIEnableState
+}
+
+type (
+  UIEnableStateOn UIEnableState = iota
+  UIEnableStatePartial
+  UIEnableStateOff
+)
+```
+
+In this model, there's NO source of truth in the model of whether a resource is
+enabled or disabled.  It's only triggered with a toggle.  The source of truth is
+internal to the controller. But interfaces can still get the "result" of being
+disabled from the UIResource aggregator.
+
+The individual controllers get to decide how to process those toggles. If you
+click the button twice before the controller responds, the controller gets to
+decide if that counts as one toggle or two toggles.
+
+NB: This might be a good thing! For resources where it takes a long time to
+start or stop a service, you probably WANT the controller to make smart
+decisions about "queued up" toggles.
+
+A slight variant on this idea: `DisableStatus` doesn't explicitly model whether
+the resource is disabled. Instead, we just see the resource has been terminated
+with a reason "Disabled". Different types of resource might model this in
+different ways (for example, Cmd might have it be a propery of its
+CmdStateTerminated and CmdStatePending fields.)
+
+### DisableSource
+
+If an API object can be disabled, then that API object should have three new
+fields: `Disabled`, `DisableSource`, and `DisableStatus`. `Disabled` and
+`DisableSource` points at the source of truth data model for whether the
+resource is disabled.
+
+Here's what that might look like:
+
+```go
+type KubernetesApplySpec struct {
+  ...
+  
+  Disabled *bool
+  DisableSource *DisableSource
+}
+
+type KubernetesApplyStatus struct {
+  ...
+  
+  DisableStatus *DisableStatus
+}
+
+type DisableSource struct {
+  ConfigMap *ConfigMapDisableSource
+}
+
+type ConfigMapDisableSource struct {
+  // The name of the ConfigMap
+  Name string
+    
+  // The key where the enable/disable state is stored.
+  Key string
+  
+  // Specify whether the ConfigMap must be defined.
+  Optional *bool
+}
+
+type DisableStatus struct {
+  Disabled bool
+  LastUpdateTime metav1.Time
+  Reason string
+}
+```
+
+The DisableStatus for a set of objects would be aggregated into UIResource like
+in the previous proposal.
+
+There's some prior art here: I've borrowed this pattern from Kubernetes, where
+you can specify the environment variables for a container with `Env` and `EnvFrom`.
+
+https://pkg.go.dev/k8s.io/api/core/v1#Container
+
+You could imagine starting out by just implementing one of the fields
+(either `Disabled` or `DisableSource` would be fine).
+
+For the initial implementation, I would have the Tiltfile create ConfigMaps for
+each UIResource, and wire up the each object's DisableSource to the correct
+ConfigMap.
```

Comment:
here's a good thread on it: https://github.com/kubernetes/kubernetes/issues/22368

my understanding is that there's broad support for a "pick up changes to ConfigMaps" feature, but also broad disagreement about the details of how it would work in the particulars (e.g., if you're expecting the Pod to restart, or the Volume to live-reload without notifying anyone, or something else)

Reply:
(hence why there are different k8s plugins that add this functionality, rather than core Kubernetes taking a position on it)